sqlgen/db_test.go

```diff
@@ -9,6 +9,45 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestPanicOnNoIndex(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("We expect non-indexed queries to panic.")
+		}
+	}()
+
+	tdb, db, err := setup()
+	assert.NoError(t, err)
+	defer tdb.Close()
+
+	db, err = db.WithPanicOnNoIndex()
+
+	// A second set will error
+	_, err = db.WithPanicOnNoIndex()
```

Comment:
ooc why do we want to panic if we set this a second time? why not just fail silently since there is no change?

Reply:
It errors, not panics, and I tend to agree with you on silently running, but other functions similar to this in the file error out. It does make sense to some extent since it indicates an odd db setup from the caller.